status-reporter/status_reporter/rabbitmq_reporter.py

```diff
@@ -13,15 +13,101 @@
 #  * See the License for the specific language governing permissions and
 #  * limitations under the License.
 
-from cloudify.cluster_status import CloudifyNodeType
+import yaml
+import json
+import requests
 
-from .status_reporter import Reporter
+from cloudify.utils import LocalCommandRunner
+from cloudify.exceptions import CommandExecutionException
+from cloudify.cluster_status import (CloudifyNodeType, ServiceStatus,
+                                     NodeServiceStatus)
 
+from .utils import get_systemd_services, get_node_status, get_service_status
+from .status_reporter import Reporter, logger
 
-def collect_status(reporter_credentials):
-    return {}
+
+class RabbitMQReporter(Reporter):
+    def __init__(self):
+        super(RabbitMQReporter, self).__init__(CloudifyNodeType.BROKER)
+
+    def _collect_status(self):
+        report = dict.fromkeys(['status', 'services'])
+        rabbitmq_credentials = _get_rabbitmq_credentials()
+        _check_rabbitmq_service(report)
+        if report['services']['RabbitMQ']['status'] == \
+                NodeServiceStatus.INACTIVE:  # rabbitmq service is down
+            _fail_tests(report)
+        else:
+            _check_and_report_node_status(report, rabbitmq_credentials)
+            _check_and_report_rabbitmq_cluster_status(report)
+        _update_report_status(report)
+        return report['status'], report['services']
+
+
+def _get_rabbitmq_credentials():
+    with open('/etc/cloudify/config.yaml') as config_file:
```

Comment:
You can use instead the function from the utils: `read_from_yaml_file` and catch the exception.

Reply:
Also I see I'll also need to get info from the config file so we can create a util function.
And const `'/etc/cloudify/config.yaml'`